transmission.go

```diff
@@ -259,24 +252,32 @@ func (b *batchAgg) fireBatch(events []*Event) {
 	sd.Count("messages_sent", numEncoded)
 	defer resp.Body.Close()
 
-	// decode the responses
-	batchResponses := []Response{}
-	err = json.NewDecoder(resp.Body).Decode(&batchResponses)
-	if err != nil {
-		// if we can't decode the responses, just error out all of them
-		sd.Increment("response_decode_errors")
+	if resp.StatusCode != http.StatusOK {
+		sd.Increment("send_errors")
+		body, _ := ioutil.ReadAll(resp.Body)
 		for _, ev := range events {
 			if ev != nil {
 				b.enqueueResponse(Response{
-					Duration: dur,
-					Metadata: ev.Metadata,
-					Err:      err,
+					StatusCode: resp.StatusCode,
+					Body:       body,
```

Comment:
Opinion on whether we should be setting `Err` in addition to `Body` here?

Reply:
Hmm, good question. For now, I'd vote to just set the status code and body. We can add an error type later if it proves too confusing. The only relevant client code that I could find ([here](https://github.com/honeycombio/honeytail/blob/1824c537e0ff3b38f3140a8fb7df1b3953a89c9f/leash.go#L497), probably not working correctly before this patch) only looks at the status code.